### PR TITLE
add functionality to search by categories

### PR DESCRIPTION
**Pull request by DavidSerranoReyes**
please check if its ok, i check and works great to be able search by categories, but need to make change in backend to call it.

### CODE REVIEW

File: backend/app/routers/idioms.py [hunk 1]
```diff
@@ -15,12 +15,26 @@
...
+@router.get("/categories", response_model=list[str])
+async def get_categories(db: SessionDep) -> list[str]:
+    """Get all unique categories from the database"""
+    result = db.execute(
+        sql_text(
+            "SELECT DISTINCT unnest(context_diversity) as context_diversity "
+            "FROM idioms ORDER BY context_diversity"
+        )
+    )
+    # print( len([row[0] for row in result.fetchall()]) )
```
Comment (alvarotroya) [inline backend/app/routers/idioms.py:27]: let's remove this

File: backend/tests/integration/routers/test_idioms.py [hunk 2]
```diff
@@ -67,6 +67,137 @@ def test_get_idioms_sort_imagery(test_server, idioms_test_data):
...
+    assert response.status_code == 200
+
+    categories = response.json()
+    assert isinstance(categories, list)
+    assert len(categories) > 0
+
+    # Collect all expected categories from test data
+    expected_categories = set()
+    for idiom in idioms_test_data.values():
+        expected_categories.update(idiom.context_diversity)
```
Comment (alvarotroya) [inline backend/tests/integration/routers/test_idioms.py:81]: this can be `expected_categories = sorted(set(idioms_test_data.values))`

which takes care of the sorting, you might need to cast to list first though, afaik sets are not sortable, might be that that changed in newer python versions, did you try running the tests? `uv run pytest` should do it

File: backend/tests/integration/routers/test_idioms.py [hunk 3]
```diff
@@ -67,6 +67,137 @@ def test_get_idioms_sort_imagery(test_server, idioms_test_data):
...
+
+    expected_categories_sorted = sorted(expected_categories)
+
+    # Verify categories are returned in alphabetical order
+    assert categories == expected_categories_sorted
+
+    # Verify no duplicates
+    assert len(categories) == len(set(categories))
```
Comment (alvarotroya) [inline backend/tests/integration/routers/test_idioms.py:89]: we don't need this as we are already testing against a set in line 86

File: backend/tests/integration/routers/test_idioms.py [hunk 4]
```diff
@@ -67,6 +67,137 @@ def test_get_idioms_sort_imagery(test_server, idioms_test_data):
...
+
+
+def test_get_categories_returns_expected_categories(test_server):
+    response = test_server.get("/idioms/categories")
+    assert response.status_code == 200
+
+    categories = response.json()
```
Comment (alvarotroya) [inline backend/tests/integration/routers/test_idioms.py:96]: we don't need this test, we are testing this above as well.

File: backend/tests/integration/routers/test_idioms.py [hunk 5]
```diff
@@ -67,6 +67,137 @@ def test_get_idioms_sort_imagery(test_server, idioms_test_data):
...
+
+    assert_idioms(actual_idioms, expected_idioms)
+    assert len(actual_idioms) > 0, (
+        "Should find at least one idiom with 'business' category"
+    )
+
+
+def test_get_idioms_filter_by_category_daily_life(test_server, idioms_test_data):
+    # Test filtering by "daily life" category
+    response = test_server.get("/idioms/?category=daily life")
```
Comment (alvarotroya) [inline backend/tests/integration/routers/test_idioms.py:136]: we don't need this test, this is a duplicate of the above one

File: backend/tests/integration/routers/test_idioms.py [hunk 6]
```diff
@@ -67,6 +67,137 @@ def test_get_idioms_sort_imagery(test_server, idioms_test_data):
...
+
+    # Verify each returned idiom meets both criteria
+    for idiom_dict in actual_idioms:
+        idiom = IdiomCreate(**idiom_dict)
+        assert "business" in idiom.context_diversity, (
+            f"Idiom '{idiom.text}' should have 'business' category"
+        )
+        assert "work" in idiom.text.lower(), (
+            f"Idiom '{idiom.text}' should contain 'work' in text"
+        )
```
Comment (alvarotroya) [inline backend/tests/integration/routers/test_idioms.py:178]: let's still use the assert idioms method, that should take care of this

**Review by alvarotroya — APPROVED** [review]
lgtm! just left some minor comments about the tests
